db/1.8/modules/ROOT/pages/database-connector-connection.adoc

```diff
@@ -0,0 +1,393 @@
+
+[start=6]
+. In the *Global Elements* view, click *Create* to open the *Choose Global Type* window.
+. In the *Filter* field write `database`, select *Database Config* and click *OK*.
+. In the *Database Config* window, for *Connection* select *Data Source Reference Connection*.
+. Click *Configure* to add the required JDBC driver and select either: +
++
+* *Use local file* +
+Install the library using a local file.
+* *Add Maven dependency* +
+Install a Maven dependency to add to the project.
++
+. For *Data source ref* add the bean id `jdbcDataSource` created in the `datasource-config.xml` bean file.
+. Click *Test Connection* to confirm that Mule can connect to the database.
+. Click *OK*.
+
+The following screenshoot shows the *Data Source Reference Connection* configuration:
+
+.Data Source Reference Connection
+image::database-datasource-connection.png[Data Source Reference Connection]
+
+In the XML editor, both the spring and database configuration looks like this:
+
+[source,xml,linenums]
+----
+<spring:config name="datasource" files="datasource-config.xml" />
+
+<db:config name="Database_Config">
+   <db:data-source-connection dataSourceRef="jdbcDataSource" />
+</db:config>
+----
+
+== Configure a Derby Connection
+
+Use the *Derby Connection* type when you want to connect to a Derby database.
+
+The following example shows how to configure a *Derby Connection* in Anypoint Studio:
+
+. In Studio, navigate to the *Global Elements* view.
+. Click *Create* to open the *Choose Global Type* view.
+. In the *Filter* field write `database`, select *Database Config* and click *OK*.
+. In the *Database Config* window, for *Connection* select *Derby Connection*.
+. Click *Configure* to add the required Derby JDBC driver and select either: +
++
+* *Add recommended library* +
+Install the recommended library.
+* *Use local file* +
+Install the library using a local file.
+* *Add Maven dependency* +
+Install a Maven dependency to add to the project.
++
+. Set *Database* to `derbyDB`, *Subsub protocol* to `directory`.
+. On the *Transactions* tab, optionally specify the transaction isolation, or XA transactions when connecting to the database.
+. On the *Advanced* tab, optionally specify connection pooling and reconnection information, including a reconnection strategy.
+. Click *Test Connection* to confirm that Mule can connect to the database.
+. Click *OK*.
+
+The following screenshot shows the *Derby Connection* configuration in Studio:
+
+.Derby Connection configuration
+image::database-derby-connection.png[Derby Connection configuration]
+
+In the XML editor, the configuration looks like this:
+[source,xml,linenums]
+----
+<db:config name="Database_Config" >
+		<db:derby-connection database="derbyDB" />
+	</db:config>
+----
+
+== Configure a Generic Connection
+
+Use the *Generic Connection* type when you want to connect to:
+
+* A database for which MuleSoft does not provide a specific connection type.
```

Comment:
List items here are not sentences, so no period (intro does not count as part of list item):
```suggestion
* A database for which MuleSoft does not provide a specific connection type
```